qemu/tests/uefishell.py

```diff
@@ -121,21 +123,28 @@ def send_command(self, command, check_result=False, interval=0.5):
 
         :param command: the command string being executed
         :param check_result: the pattern to validate output
+        :param interval: time interval between commands
+        :return if check_result is True, return matched string
         """
         logging.info("Send uefishell command: %s" % command)
         output = self.session.cmd_output_safe(command)
         time.sleep(interval)
-        #Judge if cmd is run successfully via environment variable 'lasterror'
+        # Judge if cmd is run successfully via environment variable 'lasterror'
         last_error = self.params["last_error"]
         env_var = self.session.cmd_output_safe("set")
-        if not re.search(last_error, env_var):
+        if not re.search(last_error, env_var, re.S):
             self.test.fail("Following errors appear %s when running command %s"
                            % (output, command))
         if check_result:
+            value = ""
             for result in check_result.split(", "):
-                if not re.search(result, output):
+                if not re.findall(result, output, re.S):
```

Comment:
why findall here, the first matched string will be returned, search cannot do it?

Reply:
the return value is group(0) or group(1) by re.search()  
re.findall is more easier.
e.g.
ip_str = \"\"\"FS0:\\> ifconfig -l eth0 auto
aaaaaaaa
bbbbbbbb
ipv4 address : 10.73.130.121 
host addr    : 
FE80::9834:39FF:FE31:A413/64\"\"\"

patern = "ipv4\\s+address\\s+:\\s+(\\d+.\\d+.\\d+.\\d+)"

ip1 = re.findall(patern, ip_str, re.S)[0]
print(ip1)
ip2 = re.search(patern, ip_str, re.S).group(0)
print(ip2)
ip3 = re.search(patern, ip_str, re.S).group(1)
print(ip3)

Reply:
Please be noted that '.' in the pattern is not the regular char '.' but a special char, which matches any char except a newline. 
here,  it's OK, for the 'ipv4 address :' can locate the line including the ipv4 address.

Reply:
OK, I see. thanks.